Work branch: Comments module

erastian opened the pull request:
- Add comments module.
- Fixed authorID type in ArticleCreateDTO
- Changed validation decorator for mongoDB ID fields in article-create.dto & comment-create.dto

erastian committed:
Work branch: Comments module

- Add comments module.

mdziarkach commented on src/articles/dto/article-create.dto.ts:
```diff
@@ -26,10 +26,10 @@ export class ArticleCreateDto implements Omit<Article, 'authorID' | 'id'> {
 
 	@IsDefined()
-	@IsNumber()
+	@IsString()
```
Add please custom decorator for cheching id
https://github.com/typestack/class-validator/issues/630

erastian replied:
Done

mdziarkach commented on src/comments/comments.repository.ts:
```diff
@@ -0,0 +1,85 @@
+		}); //TODO: Add paginator. Better way get this query from article?
+	}
+
+	async findArticleComments(articleID: string, published: boolean | null): Promise<Comment[]> {
```
create please 2 methods findPublishedCommentsByArticleId and findUnpublishedCommentsByArticleId

erastian replied:
Splited method in the comment repository module into 2 different methods for comments on articles for the roles of anonymous/user and moderator/admin

mdziarkach commented on src/comments/comments.controller.ts:
```diff
@@ -0,0 +1,101 @@
+		}
+	}
+
+	async findAllComments(req: Request, res: Response, next: NextFunction): Promise<void> {
```
who will use it? I don't see any scenario when we need all comments

erastian replied:
Moderator. One place for all comments with a lot of sorts and filters

erastian replied:
Decided to continue working on this endpoint with appropriate methods later

mdziarkach commented on src/comments/dto/comment-create.dto.ts:
```diff
@@ -0,0 +1,21 @@
+
+	@IsDefined()
+	@IsString()
+	articleID: string;
```
use custom middleware please for checking ids

erastian replied:
Done

erastian committed:
Work branch: Auth module

- Changed validation decorator for mongoDB ID fields in article-create.dto & comment-create.dto

erastian committed:
Work branch: Comments module

- Splited method in the comment repository module into 2 different methods for comments on articles for the roles of anonymous/user and moderator/admin
